sdk/servicebus/microsoft-azure-servicebus/src/test/java/com/microsoft/azure/servicebus/jms/SBJmsConnectionFactory.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.servicebus.jms;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSContext;
+import javax.jms.JMSException;
+import org.apache.qpid.jms.JmsConnectionFactory;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+
+public class SBJmsConnectionFactory implements ConnectionFactory {
+    private JmsConnectionFactory factory;
+
+    private SBJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
+        if (sasKeyName == null || sasKeyName == null || host == null) {
+            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
+        }
+        
+        String query = (settings == null) ? "" : settings.toQuery();
+        this.factory = new JmsConnectionFactory(sasKeyName, sasKey, "amqps://" + host + query);
+    }
+    
+    public static ConnectionFactory create(String connectionString, ServiceBusJmsConnectionFactorySettings settings) {
+        return create(new ConnectionStringBuilder(connectionString), settings);
+    }
+    
+    public static ConnectionFactory create(ConnectionStringBuilder connectionStringBuilder, ServiceBusJmsConnectionFactorySettings settings) {
```

Comment:
>create(ConnectionStringBuilder connectionStringBuilder [](start = 36, length = 54)

Do you know if this can support MSI?

Reply:
No I don't think so, SAS only

---
In reply to: [316400201](https://github.com/bainian12345/azure-sdk-for-java/pull/6#discussion_r316400201) [](ancestors = 316400201)